evalml/automl/engines/engine_base.py

```diff
@@ -0,0 +1,212 @@
+        if len(desc) > self.automl._MAX_NAME_LEN:
+            desc = desc[:self.automl._MAX_NAME_LEN - 3] + "..."
+        desc = desc.ljust(self.automl._MAX_NAME_LEN)
+
+        update_pipeline(logger,
+                        desc,
+                        len(self.automl._results['pipeline_results']) + 1,
+                        self.automl.max_iterations,
+                        self.automl._start,
+                        1 if self.automl._automl_algorithm.batch_number == 0 else self.automl._automl_algorithm.batch_number,
+                        self.automl.show_batch_output)
+
+    def _add_result_callback(self, result_callback, pipeline, evaluation_result):
+        """Calls the result callback function with the pipeline evaluation results and updates the search iteration plot.
+
+        Arguments:
+            result_callback (callable): Function called after a pipeline is finished evaluation
+            pipeline (PipelineBase): An untrained pipeline with the parameters used during training
+            evaluation_result (dict): The training results for the pipeline
+        """
+        parameters = pipeline.parameters
+        logger.debug('Adding results for pipeline {}\nparameters {}\nevaluation_results {}'.format(pipeline.name, parameters, evaluation_result))
+        result_callback(trained_pipeline=pipeline,
+                        parameters=parameters,
+                        training_time=evaluation_result['training_time'],
+                        cv_data=evaluation_result['cv_data'],
+                        cv_scores=evaluation_result['cv_scores'])
+        logger.debug('Adding results complete')
+
+        if self.automl.search_iteration_plot:
+            self.automl.search_iteration_plot.update()
+
+    def _handle_keyboard_interrupt(self, current_batch_pipelines, pipeline=None):
+        """Presents a prompt to the user asking if they want to stop the search.
+
+        Arguments:
+            current_batch_pipelines (list): Other pipelines in the batch.
+
+        Returns:
+            list: Next pipelines to search in the batch. If the user decides to stop the search,
+                an empty list will be returned.
+        """
+        leading_char = "\n"
+        start_of_loop = time.time()
+        while True:
+            choice = input(leading_char + "Do you really want to exit search (y/n)? ").strip().lower()
+            if choice == "y":
+                logger.info("Exiting AutoMLSearch.")
+                return []
+            elif choice == "n":
+                # So that the time in this loop does not count towards the time budget (if set)
+                time_in_loop = time.time() - start_of_loop
+                self.automl._start += time_in_loop
+                return [pipeline] + current_batch_pipelines if pipeline else current_batch_pipelines
+            else:
+                leading_char = ""
+
+    @staticmethod
+    def _compute_cv_scores(pipeline, automl, X, y):
```

Comment:
Since this is static, this could potentially be moved back into `automl_search` rather than stay in engines. If kept in `AutoMLSearch`, it would give it one central place for the default implementation of computing CV scores. However, keeping it in engines would make sense for reimplementations of `_compute_cv_scores`. Thoughts?

Reply:
This version of `_compute_cv_scores` shouldn't have any side effects. Dask is actually able to use this as-is since there isn't any side effects. Let's try moving `_compute_cv_scores` to `AutoMLSearch` then.